Stratis.Bitcoin/Notifications/TransactionNotificationProgress.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Notifications
+{
+    /// <summary>
+    /// Class containing elements showing the progress of transactions notifications.
+    /// </summary>
+    public class TransactionNotificationProgress
+    {
+        public TransactionNotificationProgress()
+        {
+            this.TransactionsReceived = new Dictionary<uint256, uint256>();
+        }
+
+        /// <summary>
+        /// Contains hashes of the transactions that have already been received from other nodes.
+        /// </summary>
+        public Dictionary<uint256, uint256> TransactionsReceived { get; set; }        
```

Comment:
ah yes, because this dictionary can be updated by as many nodes as are connected to the current one.